src/webgl/p5.Shader.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+var p5 = require('../core/core');
+var renderer = require('./p5.RendererGL');
+
+/**
+ * p5 Shader class
+ * @constructor
+ * @param  {String} fragShader Source code of a fragment shader as
+ * @param  {String} vertShader Source code of a vertex shader as a string
+ *
+ */
+p5.Shader = function(fragSource, vertSource){
+  //TODO: this key should be the hash of the optional fragSource and vertSource
+  this.shaderKey = 'customShader';
```

Comment:
so if I understand correctly, this.shaderKey would essentially be an id for lookup in a material hash?